### DISCUSSION TITLE
Suggestions: Fix/simplify left click and shift click of Fairy Rings, Spiritual Fairy Tree, and Portal Nexus

### DISCUSSION THREAD

**Discussion by cow1337killer3** [runelite/runelite#15793] [Ideas]
1. There's already a "Fairy Ring" option under Objects, which is great. But it'd be great if there was a second one for setting the Shift Click of Fairy Rings. I wanted, for every Fairy Ring, to be able to left click "Configure" and shift click "Last Destination". To do this currently, you have to travel to every single Fairy Ring in the game and manually set the shift click swap. This is because there is a separate Fairy Ring object for every single "Last-destination (XXX)" possibility. Whenever you travel to CKR for example, it swaps all Fairy objects in the game to the specific Fairy Ring object that has "Last-destination (CKR)" as a menu option. It would be nice to have this 2nd shift click option in the plugin settings so we don't have to manually set every single Fairy Ring.

2. The Spiritual Fairy Tree object works like the Fairy Ring object. There's a separate object for every possible "Last-destination". So even if you go to every single Fairy Ring in the game and manually set the shift click option, you again have to do the same for your Spiritual Fairy Tree, but with an extra stop at your POH between each location. This could be fixed in the same manner as the Fairy Rings. Have 2 "Spiritual Fairy Tree" options under Objects, one for left click, and one for shift click. This would also solve # 3:

3. (Solved by # 2) Additionally, changing the existing "Fairy Ring" option under Objects interferes with the POH Spiritual Fairy Tree options. I want the "Tree" option to be the default left click. When I set "Fairy Ring" under Objects to "Configure", it also changes the Spiritual Fairy Tree to this. And for some reason there's no menu entry for "Swap left click Tree", so I can't manually set it. Because of this, to get things exactly how I want, I had to just set "Fairy Ring" under Objects to "Off", so "Tree" would be the default left click option, and then I again had to travel to every Fairy Ring in the game a 2nd time and manually set the left click to "Configure".

4. There's a similar issue with the Portal Nexus. This happens regardless if the "Portal Nexus" checkbox under Objects is enabled or not. I want to be able to left click it to travel to the "quick teleport" you set through the "Configuration" option, and I want to be able to shift click it to get the "Teleport Menu". This can be manually set, but then if you change your "quick teleport", it'll have to be manually set again. Like the fairy rings, there's a separate Portal Nexus object for every possible "quick teleport" option available. This problem could be solved with the same solution as suggested previously. Have 2 "Portal Nexus" options under Objects, one for left click, and one for shift click.

5. I don't think this actually matters, but the Fairy Ring in Zanaris is also its own separate set of objects (I think it had to have its own separate objects just to remove the "Zanaris" menu option). So if you really wanted to customize left click and shift click for it, you'd again have to run through every possible Fairy Ring location, going back to Zanaris between each one and manually setting the menu swaps. I don't think this would actually be useful for anyone though, so it doesn't need its own dedicated options.

In summary:
1: "Fairy Ring left click" and "Fairy Ring shift click" options.
2/3: "Spiritual Fairy Tree left click" and "Spiritual Fairy Tree shift click" options.
4: "Portal Nexus left click" and "Portal Nexus shift click" options.

This may seem a little excessive, but I honestly think it's the best solution. And I believe it's needed because these are objects that players use very frequently and would find a lot of benefit from being able to properly customize. 

It'd be nice if we could get these improvements!

**Comment by Allu71** [comment]
Yeah my left click configure option on fairy rings keeps randomly changing to last destination
